flang/lib/Optimizer/Builder/Runtime/Command.cpp

```diff
@@ -0,0 +1,66 @@
+//===-- Command.cpp -- generate assignment runtime API calls ---------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "flang/Optimizer/Builder/Runtime/Command.h"
+#include "flang/Optimizer/Builder/FIRBuilder.h"
+#include "flang/Optimizer/Builder/Runtime/RTBuilder.h"
+#include "flang/Optimizer/Builder/BoxValue.h"
+#include "flang/Runtime/command.h"
+#include "llvm/ADT/Optional.h"
+using namespace Fortran::runtime;
+
+mlir::Value fir::runtime::genCommandArgumentCount(fir::FirOpBuilder & builder, mlir::Location loc)
+{
+    auto argumentCountFunc = fir::runtime::getRuntimeFunc<mkRTKey(ArgumentCount)>(loc, builder);
+    return builder.create<fir::CallOp>(loc, argumentCountFunc).getResult(0);
+}
+
+// GET_COMMAND_ARGUMENT intrinsic is split between 2 functions in implementation; ArgumentValue and ArgumentLength.
+// So we handle each seperately.
```

Comment:
nit: doxygen? in the header file, of course.

Reply:
Done.